crates/kubelet/src/container/status.rs

```diff
@@ -82,3 +85,90 @@ impl Status {
         }
     }
 }
+
+/// Patch a single container's status
+pub async fn patch_container_status(
```

Comment:
This relies on container statuses matching the containers in `Pod.containers` and `Pod.init_containers`, in the same order. I believe that this would be best practice.